packages/core/src/components/tag/tag.tsx

```diff
@@ -50,6 +50,13 @@ export class Tag extends React.Component<ITagProps, {}> {
             </span>
         );
     }
+
+    private onRemoveClick = (e: React.MouseEvent<HTMLButtonElement>) => {
+        const { onRemove } = this.props;
+        if (onRemove) {
+            onRemove(e, this.props);
```

Comment:
since `onRemove` is effectively `Function | undefined`, the typechecker should guarantee this to be safe. this callback only actually occurs if `isFunction(onRemove)` from line 42, though that isn't expressed in the types here. what does `safeInvoke` provide that the typechecker cannot? happy to rewrite this whole function as:

```typescript
private onRemoveClick = (e: React.MouseEvent<HTMLButtonElement>) => {
  const { onRemove } = this.props;
  safeInvoke(onRemove, e, this.props);
}
```

if preferred, just seems slightly more complicated for no gain.